presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotBrokerPageSource.java

```diff
@@ -216,11 +217,36 @@ public Page getNextPage()
             List<Type> expectedTypes = columnHandles.stream()
                     .map(PinotColumnHandle::getDataType)
                     .collect(Collectors.toList());
+            List<Integer> columnPositionsInGroupByResults = new ArrayList<>(columnHandles.size());
             PageBuilder pageBuilder = new PageBuilder(expectedTypes);
             ImmutableList.Builder<BlockBuilder> columnBlockBuilders = ImmutableList.builder();
             ImmutableList.Builder<Type> columnTypes = ImmutableList.builder();
+            int metricIdx = 0;
+            for (int i = 0; i < columnHandles.size(); i++) {
```

Comment:
I am wondering if this state computation can be done at the time of PQL generation ?

Having this here is somewhat deviant from the current flow of the code and I am not sure if it strictly needs to be done here: The current design of the code is to prepare all the state needed in the PQL generation phase and then simply follow that in the page source. We don't attempt to rederive the state needed in the page source. 

We have more context during PQL generation to be able to do this reliably.